libs/model-assessment/src/lib/ModelAssessmentDashboard/Controls/ModelOverview/ChartConfigurationFlyout.tsx

```diff
@@ -0,0 +1,213 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import {
+  defaultModelAssessmentContext,
+  ErrorCohort,
+  ModelAssessmentContext
+} from "@responsible-ai/core-ui";
+import { localization } from "@responsible-ai/localization";
+import {
+  Dropdown,
+  Stack,
+  IDropdownOption,
+  Panel
+} from "office-ui-fabric-react";
+import React from "react";
+
+interface IChartConfigurationFlyoutProps {
+  isOpen: boolean;
+  onDismissFlyout: () => void;
+  datasetCohorts: ErrorCohort[];
+  featureBasedCohorts: ErrorCohort[];
+  selectedDatasetCohorts: number[];
+  selectedFeatureBasedCohorts: number[];
+  updateDatasetCohortSelection: (selectedDatasetCohorts: number[]) => void;
+  updateFeatureBasedCohortSelection: (selectedDatasetCohorts: number[]) => void;
+}
+
+class IChartConfigurationFlyoutState {}
+
+const selectAllOptionKey = "selectAll";
+
+export class ChartConfigurationFlyout extends React.Component<
+  IChartConfigurationFlyoutProps,
+  IChartConfigurationFlyoutState
+> {
+  public static contextType = ModelAssessmentContext;
+  public context: React.ContextType<typeof ModelAssessmentContext> =
+    defaultModelAssessmentContext;
+
+  componentDidUpdate(
+    prevProps: IChartConfigurationFlyoutProps,
+    _prevState: IChartConfigurationFlyoutState
+  ) {
+    if (
+      this.props.featureBasedCohorts.length !==
+        prevProps.featureBasedCohorts.length ||
+      this.props.featureBasedCohorts.some(
+        (value, index) =>
+          value.cohort.name !== prevProps.featureBasedCohorts[index].cohort.name
+      )
+    ) {
+      // Feature-based cohorts changed, so update state accordingly.
+      // If there are at least 10 selected dataset cohorts
+      // don't select feature-based cohorts by default
+      this.props.updateFeatureBasedCohortSelection(
+        this.props.selectedDatasetCohorts.length >= 10
+          ? []
+          : this.getAllFeatureBasedCohorts()
+      );
+    }
+  }
+
+  public render(): React.ReactNode {
+    const selectAllOption = {
+      key: selectAllOptionKey,
+      text: localization.ModelAssessment.ModelOverview.selectAllCohortsOption
+    };
+    const datasetCohortOptions = this.props.datasetCohorts.map(
+      (cohort, index) => {
+        return { key: index.toString(), text: cohort.cohort.name };
+      }
+    );
+    const featureBasedCohortOptions = this.props.featureBasedCohorts.map(

```

Comment:
I think this map over cohort/index to return key/text could be moved into a very small helper function and re-used above on line 69 as well

Reply:
I wonder if we need some sort of common CohortUtils file in core which would have such simple shared utilities for cohorts

Reply:
It seems to be fairly custom. If something like this keeps reoccurring we should do it, though.

Reply:
I incorporated the helper, of course.